rsmtool/utils/metrics.py

```diff
@@ -356,32 +356,44 @@ def difference_of_standardized_means(y_true_observed,
     y_pred_population_params = [population_y_pred_mn,
                                 population_y_pred_sd]
 
-    if any(y_true_observed_population_params) and not all(y_true_observed_population_params):
+    if len([param for param in y_true_observed_population_params
+            if param is None]) == 1:
         raise ValueError('You must pass both `population_y_true_observed_mn` and '
                          '`population_y_true_observed_sd` or neither.')
 
-    if any(y_pred_population_params) and not all(y_pred_population_params):
+    if len([param for param in y_pred_population_params
+            if param is None]) == 1:
         raise ValueError('You must pass both `population_y_pred_mn` and '
                          '`population_y_pred_sd` or neither.')
 
     warning_msg = ('You did not pass population mean and std. for `{}`; '
                    'thus, the calculated z-scores will be zero.')
 
     # if the population means and standard deviations were not provided, calculate from the data
-    if not population_y_true_observed_mn or not population_y_true_observed_sd:
+    # We only check for mean since the function requires
+    # both of these to be set of both to be None
+    if population_y_true_observed_mn is None:
 
         warnings.warn(warning_msg.format('y_true_observed'))
         (population_y_true_observed_sd,
          population_y_true_observed_mn) = (np.std(y_true_observed, ddof=ddof),
                                            np.mean(y_true_observed))
 
-    if not population_y_pred_mn or not population_y_pred_sd:
+    if population_y_pred_mn is None:
 
         warnings.warn(warning_msg.format('y_pred'))
         (population_y_pred_sd,
          population_y_pred_mn) = (np.std(y_pred, ddof=ddof),
                                   np.mean(y_pred))
 
+    # if any of the standard deviations equal zero
+    # raise a warning and return None
+    if population_y_pred_sd == 0 or population_y_true_observed_sd == 0:
+        warnings.warn("Population standard deviations for the computation of "
+                      "DSM is zero. No value will be computed.")
+        return None
```

Comment:
Should we just raise an error here instead of returning `None`? How does a return value of `None` get handled in the report?

Reply:
I'd argue against it: we allow constant system scores for system evaluation so there is no reason to raise an error for this particular metrics given that we already return `None` for `R2` and `r` and other metrics that require non-zero variance. 

Let me add a regression test for this case though to make sure nothing in the report fails.

Reply:
Sure, that makes sense. Thanks!

Reply:
The report correctly shows either `0` or `nan` depending on the metrics: e.g. for `R2` and `kappa` we report `0` and for correlation we report `nan`. 

Having said that, I had to switch `sd == 0` to `np.islose(sd, 0)` to avoid ghost DSM where there should be none.